service/mcs/rest/src/main/java/eu/europeana/cloud/service/mcs/rest/RepresentationRevisionsResource.java

```diff
@@ -73,14 +74,33 @@ public Representation getRepresentationRevision(@Context UriInfo uriInfo,
             DateTime utc = new DateTime(revisionTimestamp, DateTimeZone.UTC);
             revisionDate = utc.toDate();
         }
-        RepresentationRevisionResponse info = recordService.getRepresentationRevision(globalId, schema, revisionProviderId, revisionName, revisionDate);
-        Representation representation;
+        List<RepresentationRevisionResponse> info = recordService.getRepresentationRevisions(globalId, schema, revisionProviderId, revisionName, revisionDate);
+        List<Representation> representations = new ArrayList<>();
         if (info != null) {
-            representation = recordService.getRepresentation(info.getCloudId(), info.getRepresentationName(), info.getVersion());
-            EnrichUriUtil.enrich(uriInfo, representation);
+            for (RepresentationRevisionResponse representationRevisionsResource : info) {
+                Representation representation;
+                representation = recordService.getRepresentation(
+                        representationRevisionsResource.getCloudId(),
+                        representationRevisionsResource.getRepresentationName(),
+                        representationRevisionsResource.getVersion());
+                EnrichUriUtil.enrich(uriInfo, representation);
+                //
+                if (userHasAccessTo(representation)) {
+                    representations.add(representation);
+                }
+            }
         } else
             throw new RepresentationNotExistsException("No representation was found");
 
-        return representation;
+        return representations;
+    }
+
+    private boolean userHasAccessTo(Representation representation){
```

Comment:
I understand why you added this code, and I think this is correct. However, it could affect the performance since we will check every returned representation, maybe we can only  check the credentials against cloudId and that is it? it is a trade-off between efficiency and completeness!

Reply:
I wouldn't care about it too. 
It is also possible to create representation version for the cloudId that was created by someone else (from another user account), so checking credentials against cloudId only may not work in some cases.